shared/src/commonMain/kotlin/org/hyperskill/app/track_selection/details/presentation/TrackSelectionDetailsReducer.kt

```diff
@@ -118,13 +118,9 @@ internal class TrackSelectionDetailsReducer : StateReducer<State, Message, Actio
                                 isNewUserMode = state.isNewUserMode
                             )
                         else ->
-                            ViewAction.NavigateTo.Home(
-                                command = if (state.isNewUserMode) {
-                                    ViewAction.NavigateTo.Home.NavigationCommand.NewRootScreen
-                                } else {
-                                    ViewAction.NavigateTo.Home.NavigationCommand.BackTo
-                                }
-                            )
+                            // navigate to first problem onboarding
+                            // if user is new and no projects to select
+                            ViewAction.NavigateTo.FirstProblemOnboarding(isNewUserMode = true)
```

Comment:
Not covered case when a user is new and projects to select are available - after the project is selected we should modify logic and navigate to the first problem onboarding if needed (`ProjectSelectionDetailsFeature`)

Reply:
> @XanderZhu user choose track in app only if he registered there. Now users registered in app have freemium subscription and they [won't be redirected to choosing project](https://github.com/hyperskill/mobile-app/blob/01a370d73a84df6335999fd21354258735c192c9/shared/src/commonMain/kotlin/org/hyperskill/app/track_selection/details/presentation/TrackSelectionDetailsReducer.kt#L141). It means that there is no flow for redirection such users on first problem onboarding from project selection

@vladkash search for `AppFeature.Action.ViewAction.NavigateTo.TrackSelectionScreen`, we are navigating to that screen not only after the user logged in event, for example after [determining user account status](https://github.com/hyperskill/mobile-app/blob/develop/shared/src/commonMain/kotlin/org/hyperskill/app/main/presentation/AppReducer.kt#L96)

Reply:
> @XanderZhu user choose track in app only if he registered there. Now users registered in app have freemium subscription and they [won't be redirected to choosing project](https://github.com/hyperskill/mobile-app/blob/01a370d73a84df6335999fd21354258735c192c9/shared/src/commonMain/kotlin/org/hyperskill/app/track_selection/details/presentation/TrackSelectionDetailsReducer.kt#L141). It means that there is no flow for redirection such users on first problem onboarding from project selection

@vladkash search for `AppFeature.Action.ViewAction.NavigateTo.TrackSelectionScreen`, we are navigating to that screen not only after the user logged in event, for example after [determining user account status](https://github.com/hyperskill/mobile-app/blob/develop/shared/src/commonMain/kotlin/org/hyperskill/app/main/presentation/AppReducer.kt#L96)

Reply:
@ivan-magda @vladkash 
In this case, I suggest moving navigation to the FirstProblemOnboarding to the AppFeature, to not spread this logic through the features. 
After the track or project selection, corresponding features will notify AppFeaute, that the track or project selection is done. And then, AppFeature can determine what to do next.

Reply:
@ivan-magda @XanderZhu 

>@vladkash search for AppFeature.Action.ViewAction.NavigateTo.TrackSelectionScreen, we are navigating to that screen not only after the user logged in event, for example after [determining user account status](https://github.com/hyperskill/mobile-app/blob/develop/shared/src/commonMain/kotlin/org/hyperskill/app/main/presentation/AppReducer.kt#L96) in that case user will be in new user mode, will skip onboarding into notifications and directly navigate to the first problem onboarding

Yes, we navigate to that screen not only after logging in. My point is that now user selects track if he is new on platform, otherwise he always has some selected track (the last one). And every such user will login into the app at the first time, after that login we will show onboarding to notifications for him and then navigate to track selection and then to first problem onboarding.

Implementing navigation to first problem onboaring in project selection feature make sense only if new users on the platform (users without track) have subscription type PREMIUM, PERSONAL or TRIAL. But now our platform has freemium model and new users have FREEMIUM subscription type. 

I tried to implement this task with minimum changes in existing features to reduce probability of bugs and errors. I think, that this implementation is normal for purposes of current task, but we should create separate task about refactoring of navigation logic in `AppFeature`, now the flow of this feature is really complex and needed to be simplified. It isn't good idea to overcomplicate solution in this task in order to deliver value of first problem onboarding to user ASAP.

Reply:
OK, we will create create separate task for navigation flow refactoring